### PR TITLE
Change over use of docker binary in Docker to use docker-py

### PR DESCRIPTION
**Pull request by a10y**
PR for #291 

@percyliang

### CODE REVIEW

**Comment by a10y** [comment]
Note: I've only tested this on docker 1.9.1, and seems to be working fine. I'm not certain if the features of the Docker Remote API are different enough across the recent versions of docker that it would lock us into using the most recent.

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+pathtools==0.1.2
+psutil==3.3.0
+python-editor==0.5
 PyYAML==3.11
+requests==2.9.0
+six==1.10.0
 SQLAlchemy==1.0.8
-alembic==0.8.2
 watchdog==0.8.3
-# Use sckoo@stanford.edu's patched argcomplete, until pull request has been accepted
```
Comment (percyliang) [inline requirements.txt:5]: Keep comment

**Comment by percyliang** [comment]
One tangential request: if user types `--request-docker-image <tab>`, then we should complete with things that begin with `codalab/...`, since novice users will want sane defaults here.  But if the user starts typing something that doesn't start with codalab, we can still do the completion as is done now.

**Comment by a10y** [comment]
Last commit in, this PR should be GTG

File: codalab/lib/completers.py [hunk 2]
```diff
@@ -183,8 +183,10 @@ class DockerImagesCompleter(CodaLabCompleter):
     Completes names of Docker images available on DockerHub.
     """
     def __call__(self, prefix, action=None, parsed_args=None):
-        def handle_err(errcode, stderr):
-            warn("Error(%d): %s" % (errcode, stderr))
+        def handle_err(exc):
+            warn("Error: %s" % exc)
```
Comment (percyliang) [inline codalab/lib/completers.py:187]: nit: use single quotes

**Comment by percyliang** [comment]
LGTM!

**Pull request merged**